src/goto-symex/path_storage.h

```diff
@@ -90,11 +90,24 @@ class path_storaget
   /// Counter for nondet objects, which require unique names
   symex_nondet_generatort build_symex_nondet;
 
+  /// Provide a unique index for a given \p id, starting from \p minimum_index.
+  std::size_t get_unique_index(const irep_idt &id, std::size_t minimum_index)
+  {
+    auto entry = unique_index_map.emplace(id, minimum_index);
+
+    if(!entry.second)
+      entry.first->second = std::max(entry.first->second + 1, minimum_index);
```

Comment:
It looks like we could avoid the `if` condition if the function is always called with the same minimum index, I don't know if that's always the case.

Reply:
No, even right now we (intentionally) use `0` for L1 indices when setting up new threads, but elsewhere use `1`. And also that facility is rather general, we could as well use it for non-det counters any maybe others.